pusher/authentication_client.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import (
+    print_function,
+    unicode_literals,
+    absolute_import,
+    division)
+
+from pusher.util import (
+    ensure_text,
+    validate_channel,
+    validate_socket_id,
+    channel_name_re)
+
+from pusher.client import Client
+from pusher.http import GET, POST, Request, request_method
+from pusher.signature import sign, verify
+
+import collections
+import hashlib
+import json
+import os
+import re
+import six
+import time
+
+
+class AuthenticationClient(Client):
+    def __init__(
+            self, app_id, key, secret, ssl=True, host=None, port=None,
+            timeout=5, cluster=None, json_encoder=None, json_decoder=None,
+            backend=None, **backend_options):
+        super(AuthenticationClient, self).__init__(
+            app_id, key, secret, ssl, host, port, timeout, cluster,
+            json_encoder, json_decoder, backend, **backend_options)
+
+        if host:
+            self._host = ensure_text(host, "host")
+
+        elif cluster:
+            self._host = (
+                six.text_type("api-%s.pusher.com") %
+                ensure_text(cluster, "cluster"))
+        else:
+            self._host = six.text_type("api.pusherapp.com")
+
+
+    def authenticate(self, channel, socket_id, custom_data=None):
+        """Used to generate delegated client subscription token.
+
+        :param channel: name of the channel to authorize subscription to
+        :param socket_id: id of the socket that requires authorization
+        :param custom_data: used on presence channels to provide user info
+        """
```

Comment:
Can you document what this method returns? It seems to be returning the auth token. If that's the case, it may be that the method's name is not very good.